If expression with different types in branches results in invalid C

**V version:** https://github.com/vlang/v/commit/86d95fcd22b05aa1cd6fdba252da6b9c150277f1
**OS:** Arch Linux

**What did you do?**
Tried to compile
```v
fn main() {
    ch := `\0`
    got := if true {
        'end of line'
    } else {ch}
    println(got)
}
```

**What did you expect to see?**
A proper compiler error message because string and byte are different types
 
**What did you see instead?**
```
dian@pc ~/p/vtest> v -debug run test2.v
C compiler=cc
/home/dian/projects/vtest/test2.tmp.c: In function ‘main’:
/home/dian/projects/vtest/test2.tmp.c:3515:50: error: type mismatch in conditional expression
 3515 | byte got= ( 1 ) ? ( tos2((byte*)"end of line") ) : ( ch ) ;
      |                                                  ^
V error: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -3073,9 +3073,11 @@ fn (p mut Parser) if_st(is_expr bool, elif_depth int) string {
 		}
 		p.check(.lcbr)
 		// statements() returns the type of the last statement
+		first_typ := typ
 		typ = p.statements()
 		p.inside_if_expr = false
 		if is_expr {
+			p.check_types(first_typ, typ)
 			p.gen(strings.repeat(`)`, elif_depth + 1))
 		}
 		else_returns := p.returns

